scripts/scil_tractogram_add_dps.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+""" Add information to each streamline from a file. Can be for example
+SIFT2 weights, processing information, bundle IDs, etc.
+
+Output must be a .trk otherwise the data will be lost.
+"""
+
+import argparse
+import logging
+
+from dipy.io.streamline import save_tractogram
+import numpy as np
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_reference_arg,
+                             add_verbose_arg,
+                             assert_inputs_exist,
+                             assert_outputs_exist,
+                             load_matrix_in_any_format)
+
+
+def _build_arg_parser():
+
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_tractogram',
+                   help='Input tractogram (.trk or .tck).')
+    p.add_argument('dps_file',
+                   help='File containing the data to add to streamlines.')
+    p.add_argument('dps_key',
+                   help='Where to store the data in the tractogram.')
+    p.add_argument('out_tractogram',
+                   help='Output tractogram (.trk or .tck).')
+
+    add_reference_arg(p)
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+    logging.getLogger().setLevel(logging.getLevelName(args.verbose))
+
+    # I/O assertions
+    assert_inputs_exist(parser, [args.in_tractogram, args.dps_file],
+                        args.reference)
+    assert_outputs_exist(parser, args, args.out_tractogram)
+
```

Comment:
assert output is trk otherwise either you choose to raise an error or just a warning, your call.